### PR TITLE
[docs] Config Plugins: update code blocks, align with writing styleguide

### PR DESCRIPTION
**Pull request by Simek**
# Why

Had to check out this page while working on latest `build-properties` API docs update, and it looks a bit neglected.

# How

Update most of the code block usages on the page so they include file name or block title, apply some of the rules from our writing guide, break lengthy lines for the editor convenience.

# Test Plan

The changes to the page have been verified by running docs app locally.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md).

### CODE REVIEW

File: docs/pages/guides/config-plugins.mdx [hunk 1]
```diff
@@ -523,21 +539,23 @@ Use the following dependencies in a library that provides a config plugin:
 ```
 
 - You may update the exact version of `expo` to build against a specific version.
-- For simple config plugins that depend on core, stable APIs, such as a plugin that only modifies **Info.plist** or **AndroidManifest.xml**, you can use a loose dependency such as in the example above.
+- For simple config plugins that depend on core, stable APIs, such as a plugin that only modifies **AndroidManifest.xml** ot **Info.plist**, you can use a loose dependency such as in the example above.
```
Comment (amandeepmittal) [inline docs/pages/guides/config-plugins.mdx:542]:
```suggestion
- For simple config plugins that depend on core, stable APIs, such as a plugin that only modifies **AndroidManifest.xml** or **Info.plist**, you can use a loose dependency such as in the example above.
```

File: docs/pages/guides/config-plugins.mdx [hunk 2]
```diff
@@ -523,21 +539,23 @@ Use the following dependencies in a library that provides a config plugin:
...
+const { /* @hide ...*//* @end */ } = require('expo/config-plugins');
+const { /* @hide ...*//* @end */ } = require('expo/config');
 ```
 
 Importing through the `expo` package ensures that you are using the version of the `expo/config-plugins` and `expo/config` packages that are depended on by the `expo` package.
 
-If you do not import the package through the `expo` re-export in this way, you may accidentally be importing an incompatible version (depending on the implementation details of module hoisting in the package manager used by the developer consuming the module) or be unable to import the module at all (if using "plug and play" features of a package manager like Yarn Berry or pnpm).
+If you do not import the package through the `expo` re-export in this way, you may accidentally be importing an incompatible version
+(depending on the implementation details of module hoisting in the package manager used by the developer consuming the module) or be unable to import the module at all
+(if using "plug and play" features of a package manager like Yarn Berry or pnpm).
```
Comment (amandeepmittal) [inline docs/pages/guides/config-plugins.mdx:558]:
```suggestion
If you do not import the package through the `expo` re-export in this way, you may accidentally be importing an incompatible version
(depending on the implementation details of module hoisting in the package manager used by the developer consuming the module) or be unable to import the module at all
(if using "plug and play" features of a package manager such as Yarn Berry or pnpm).
```

File: docs/pages/guides/config-plugins.mdx [hunk 3]
```diff
@@ -589,11 +608,14 @@ If you aren't comfortable with setting up a monorepo, you can try manually runni
 
 ### Modifying the AndroidManifest.xml
 
-Packages should attempt to use the built-in **AndroidManifest.xml** [merging system](https://android-doc.github.io/tools/building/manifest-merge.html) before using a config plugin. This can be used for static, non-optional features like permissions. This will ensure features are merged during build-time and not prebuild-time, which minimizes the possibility of users forgetting to prebuild. The drawback is that users cannot use [introspection](#introspection) to preview the changes and debug any potential issues.
+Packages should attempt to use the built-in **AndroidManifest.xml** [merging system](https://android-doc.github.io/tools/building/manifest-merge.html)
```
Comment (amandeepmittal) [inline docs/pages/guides/config-plugins.mdx:611]:
```suggestion
Packages should attempt to use the built-in **AndroidManifest.xml** [merging system](https://developer.android.com/studio/build/manage-manifests)
```

The link above doesn't work anymore. We should replace it with the official Android docs.

File: docs/pages/guides/config-plugins.mdx [hunk 4]
```diff
@@ -671,7 +692,8 @@ export const withCustomConfig: ConfigPlugin<string> = (config, id) => {
 
 ### Modifying the iOS Podfile
 
-The iOS Podfile is the config file for CocoaPods, the dependency manager on iOS. Think of Podfile like package.json but for iOS. The Podfile is a ruby file (application code), which means you **cannot** safely modify it from Expo config plugins, and should opt towards another approach, like Expo Autolinking hooks (citation needed).
+The iOS Podfile is the config file for CocoaPods, the dependency manager on iOS. Think of Podfile like package.json but for iOS.
```
Comment (amandeepmittal) [inline docs/pages/guides/config-plugins.mdx:695]:
```suggestion
The iOS **Podfile** is the config file for CocoaPods, the dependency manager on iOS. It is similar to **package.json** for iOS.
```

File: docs/pages/guides/config-plugins.mdx [hunk 5]
```diff
@@ -671,7 +692,8 @@ export const withCustomConfig: ConfigPlugin<string> = (config, id) => {
...
+The Podfile is a ruby file (application code), which means you **cannot** safely modify it from Expo config plugins, and should opt towards another approach, like Expo Autolinking hooks (citation needed).
```
Comment (amandeepmittal) [inline docs/pages/guides/config-plugins.mdx:696]:
```suggestion
The **Podfile** is a ruby file (application code), which means you **cannot** safely modify it from Expo config plugins and should opt for another approach, such as [Expo Autolinking](modules/autolinking) hooks.
```
I think we can remove the "citation needed" part and link Autolinking docs here.

File: docs/pages/guides/config-plugins.mdx [hunk 6]
```diff
@@ -710,24 +735,33 @@ export default createRunOncePlugin(
...
-- **Naming conventions**: Use `withFeatureName` if cross-platform. If the plugin is platform specific, use a camel case naming with the platform right after “with”. Ex; `withIosSplash`, `withAndroidSplash`. There is no universally agreed upon casing for `iOS` in camel cased identifiers, we prefer this style and suggest using it for your config plugins too.
-- **Leverage built-in plugins**: Account for built-in plugins from the [prebuild config](https://github.com/expo/expo-cli/blob/master/packages/prebuild-config/src/plugins/withDefaultPlugins.ts). Some features are included for historical reasons, like the ability to automatically copy and link [Google services files](https://github.com/expo/expo-cli/blob/3a0ef962a27525a0fe4b7e5567fb7b3fb18ec786/packages/config-plugins/src/ios/Google.ts#L15) defined in the Expo config. If there is overlap, then maybe recommend the user uses the built-in types to keep your plugin as simple as possible.
+- **Naming conventions**: Use `withFeatureName` if cross-platform. If the plugin is platform specific, use a camel case naming with the platform right after “with”. Ex; `withIosSplash`, `withAndroidSplash`.
```
Comment (amandeepmittal) [inline docs/pages/guides/config-plugins.mdx:743]:
```suggestion
- **Naming conventions**: Use `withFeatureName` if cross-platform. If the plugin is platform specific, use a camel case naming with the platform right after “with”. For example, `withAndroidSplash`, `withIosSplash`.
```

File: docs/pages/guides/config-plugins.mdx [hunk 7]
```diff
@@ -710,24 +735,33 @@ export default createRunOncePlugin(
...
-- **Unit test your plugin**: Write Jest tests for complex modifications. If your plugin requires access to the filesystem, use a mock system (we strongly recommend [`memfs`][memfs]), you can see examples of this in the [`expo-notifications`](https://github.com/expo/expo/blob/fc3fb2e81ad3a62332fa1ba6956c1df1c3186464/packages/expo-notifications/plugin/src/__tests__/withNotificationsAndroid-test.ts#L34) plugin tests.
+- **Unit test your plugin**: Write Jest tests for complex modifications. If your plugin requires access to the filesystem,
+  use a mock system (we strongly recommend [`memfs`][memfs]), you can see examples of this in the [`expo-notifications`](https://github.com/expo/expo/blob/fc3fb2e81ad3a62332fa1ba6956c1df1c3186464/packages/expo-notifications/plugin/src/__tests__/withNotificationsAndroid-test.ts#L34) plugin tests.
   - Notice the root [**/**mocks\*\*\*\*](https://github.com/expo/expo/tree/main/packages/expo-notifications/plugin/__mocks__) folder and [**plugin/jest.config.js**](https://github.com/expo/expo/tree/main/packages/expo-notifications/plugin/jest.config.js).
```
Comment (amandeepmittal) [inline docs/pages/guides/config-plugins.mdx:751]:
```suggestion
  - Notice the root [**/**mocks\*\*\*\*](https://github.com/expo/expo/tree/main/packages/expo-notifications/plugin/__mocks__) folder and [**plugin/jest.config.js**](https://github.com/expo/expo/tree/main/packages/expo-notifications/plugin/jest.config.js).
```

The "**/**mocks\*\*\*\*" isn't rendering correctly in HTML. I thought we can use `&lowerbar;&lowerbar;mocks&lowerbar;&lowerbar;` but Prettier in VSCode changes that to `\_\mocks\_\_`, which is fine I guess, since that pattern is also used in other paces.

TLDR; let's change this to `**\_\mocks\_\_**`.

File: docs/pages/guides/config-plugins.mdx [hunk 8]
```diff
@@ -710,24 +735,33 @@ export default createRunOncePlugin(
...
+When Expo SDK upgrades to a new version of React Native for instance, the template may change significantly to account for changes in React Native or new releases of Android or iOS.
 
-If your plugin is mostly using [static modifications](#static-modification) then it will work well across versions. If it's using a regular expression to transform application code, then you'll definitely want to document which Expo SDK version your plugin is intended for. Expo releases a new version quarterly (every 3 months), and there is a [beta period][expo-beta-docs] where you can test if your plugin works with the new version before it's released.
+If your plugin is mostly using [static modifications](#static-modification) then it will work well across versions.
+If it's using a regular expression to transform application code, then you'll definitely want to document which Expo SDK version your plugin is intended for.
+Expo releases a new version quarterly (every 3 months), and there is a [beta period][expo-beta-docs] where you can test if your plugin works with the new version before it's released.
```
Comment (amandeepmittal) [inline docs/pages/guides/config-plugins.mdx:764]:
```suggestion
Expo releases a new version quarterly (every 3 months), and there is a [beta period](https://github.com/expo/expo/blob/main/guides/releasing/Release%20Workflow.md#stage-5---beta-release) where you can test if your plugin works with the new version before it's released.
```

The link for "expo-beta-docs" docs doesn't work. I updated it here directly but feel free to move it down as how this link was referenced originally or we can delete that reference and stick with this pattern.

File: docs/pages/guides/config-plugins.mdx [hunk 9]
```diff
@@ -892,36 +923,43 @@ Expo CLI commands can be profiled using `EXPO_PROFILE=1`.
...
+- `ios.entitlements`
+- `ios.expoPlist`
+- `ios.podfileProperties`
 
 > Introspection only works on safe modifiers (static files like JSON, XML, plist, properties), with the exception of `ios.xcodeproj` which often requires file system changes, making it non idempotent.
 
-Introspection works by creating custom base mods that work like the default base mods, except they don't write the `modResults` to disk at the end. Instead of persisting, they save the results to the Expo config under `_internal.modResults`, followed by the name of the mod i.e. the `ios.infoPlist` mod saves to `_internal.modResults.ios.infoPlist: {}`.
+Introspection works by creating custom base mods that work like the default base mods, except they don't write the `modResults` to disk at the end.
+Instead of persisting, they save the results to the Expo config under `_internal.modResults`, followed by the name of the mod
+i.e. the `ios.infoPlist` mod saves to `_internal.modResults.ios.infoPlist: {}`.
```
Comment (amandeepmittal) [inline docs/pages/guides/config-plugins.mdx:949]:
```suggestion
such as the `ios.infoPlist` mod saves to `_internal.modResults.ios.infoPlist: {}`.
```

**Review by amandeepmittal — COMMENTED** [review]
Thanks for diving in to improve this doc 🙏 

I left some suggestions, and all of them are unrelated to the changes you have made.

File: docs/pages/guides/config-plugins.mdx (hunk 3 above)
Comment (Simek) [reply docs/pages/guides/config-plugins.mdx:611]: Great catch, thank you! 🙏

File: docs/pages/guides/config-plugins.mdx (hunk 7 above)
Comment (Simek) [reply docs/pages/guides/config-plugins.mdx]: Unfortunately, `**\_\mocks\_\_**` will also brake and result in bold text since `**` wrap, but I have pushed the version which should render correctly and hopefully be fine with Prettier.

Comment (amandeepmittal) [reply docs/pages/guides/config-plugins.mdx]: Thank you 🙏

**Review by amandeepmittal — APPROVED** [review]
